components/esp_modem/src/esp_modem_dce.cpp

```diff
@@ -27,24 +29,41 @@ bool DCE_Mode::set(DTE *dte, ModuleIf *device, Netif &netif, modem_mode m)
     case modem_mode::UNDEF:
         break;
     case modem_mode::COMMAND_MODE:
-        if (mode == modem_mode::COMMAND_MODE) {
-            return false;
-        }
-        netif.stop();
-        if (!device->set_mode(modem_mode::COMMAND_MODE)) {
-            return false;
+        {
+            if (mode == modem_mode::COMMAND_MODE) {
+                return false;
+            }
+            netif.stop();
+            SignalGroup signal;
+            dte->set_read_cb([&](uint8_t *data, size_t len) -> bool {
+                if (memchr(data, '\n', len)) {
+                    ESP_LOG_BUFFER_HEXDUMP("esp-modem: debug_data", data, len, ESP_LOG_DEBUG);
+                    const auto pass = std::list<std::string_view>({"NO CARRIER", "DISCONNECTED"});
+                    std::string_view response((char *) data, len);
```

Comment:
If `\n` is found before `len` find will pass the line.

Reply:
I'm sorry, missed this comment.

> If \n is found before len find will pass the line.

This is still acceptable, as the `\n` could be the starting character, too.